src/lib/isIBAN.js

```diff
@@ -95,14 +114,37 @@ const ibanRegexThroughCountryCode = {
  * NOTE: Permitted IBAN characters are: digits [0-9] and the 26 latin alphabetic [A-Z]
  *
  * @param {string} str - string under validation
+ * @param {object} options - object to pass the countries to be either whitelisted or blacklisted
  * @return {boolean}
  */
-function hasValidIbanFormat(str) {
+function hasValidIbanFormat(str, options) {
   // Strip white spaces and hyphens
   const strippedStr = str.replace(/[\s\-]+/gi, '').toUpperCase();
   const isoCountryCode = strippedStr.slice(0, 2).toUpperCase();
 
-  return (isoCountryCode in ibanRegexThroughCountryCode) &&
+  const isoCountryCodeInIbanRegexCodeObject = isoCountryCode in ibanRegexThroughCountryCode;
+
+  if (options.whitelist) {
+    if (!hasOnlyValidCountryCodes(options.whitelist)) {
+      throw new Error('One of the codes passed is invalid');
+    }
+
+    const isoCountryCodeInWhiteList = options.whitelist.includes(isoCountryCode);
+
+    if (!isoCountryCodeInWhiteList) {
+      throw new Error('IBAN code does not belong to one of the countries listed on whitelist!');
```

Comment:
nit: the pattern so far has not been to throw errors, just true/false. I'm not sure how to go about this one...

Reply:
I guess we can continue with the true/false approach and let the user decide how to treat this.

Reply:
Done @profnandaa